### DISCUSSION TITLE
[BUG] engine-manager looking for tls-files/tls.crt: no such file or directory

### DISCUSSION THREAD

**Discussion by SuitableProduce** [longhorn/longhorn#5370] [General]
## Describe the bug (🐛 if you encounter this issue)
Running  version v1.3.2, with ingress disabled, instance-manager pods have this message in their logs:
```
warning msg="Failed to add TLS key pair from /tls-files: open /tls-files/tlscrt: no such file or directory"
```

Looks like the templates/daemonset-sa.yaml is trying to use the secret `longhorn-grpc-tls`: https://github.com/longhorn/longhorn/blob/master/chart/templates/daemonset-sa.yaml#L93

I'm not sure how the secret `longhorn-grpc-tls` should get created.  Also, not sure what impact this is has.

## To Reproduce

1. Install longhorn chart
2. tail the instance-manager pods
3. Notice warning message

## Environment

 - Longhorn version: v1.3.2
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: RKE1, k8s v1.23.8

**Comment by derekbit** [comment]
It's a warning level message. Just ignore it if you don't plan to use mTLS.
Please refer to https://longhorn.io/docs/1.3.2/advanced-resources/security/mtls-support

**Comment by SuitableProduce** [comment]
Ahh ok sounds good!  Just wondering, what about adding something like this to the values:
```
mtls:
  enabled: true # True by default since thats how it is now.
```

Then in the `templates/deployment-sa.yaml`, you add the appropriate `{{- if .Values.mtls.enabled }}` section for the volume and volumeMount?  Just thinking that is a pretty low-effort change that just helps to clean things up a bit!

**Comment by weizhe0422** [comment]
I think it uses checking for the existence of the `longhorn-grpc-tls` secret to decide whether to enable mTLS to secure the communication. And it can reveal the mTLS is disabled since the secret is not exists. 

If adding a flag, I might think there is variety cases need be considered include if the secret is exist and the flag is valid or not.

It seems to me that current way is intuitive.

**Comment by weizhe0422** [comment]
UPDATE:  And if is failed load the secret file, the log shows up the connection was still created but w/o auth.


https://github.com/longhorn/longhorn-instance-manager/blob/3b0f9b08f806bd35f0f83e5455fb4b03de0a518d/app/cmd/start.go#L96-L111 
```golang
	if tlsDir != "" {
		tlsConfig, err = util.LoadServerTLS(
			filepath.Join(tlsDir, "ca.crt"),
			filepath.Join(tlsDir, "tls.crt"),
			filepath.Join(tlsDir, "tls.key"),
			"longhorn-backend.longhorn-system")
		if err != nil {
			logrus.WithError(err).Warnf("Failed to add TLS key pair from %v", tlsDir)
		}
	}


	if tlsConfig != nil {
		logrus.Info("Creating gRPC server with mtls auth")
	} else {
		logrus.Info("Creating gRPC server with no auth")
	}
```
